### ISSUE TITLE
READ THIS BEFORE NEW ISSUE

### ISSUE THREAD

**Issue by IceHacks** [nathan/phosphorus#1547]
Hello everyone. As you might know Phosphorus is **not working**. Here I will explain why. All of you are getting the following errors:

```
Access to XMLHttpRequest at 'https://scratch.mit.edu/projects/xxx/' from origin 'https://phosphorus.github.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

Now I will explain what this means and why it happens. On the internet there is a special security feature to keep external servers from accessing your site. This is called `CORS` or `Cross-Origin Resource Sharing`. Unless otherwise set, `CORS` will block requests coming from anywhere besides the host (ex: "scratch.mit.edu"). You can not override this. So **the function of getting the code from Scratch will not work**, thus making the whole thing not work. The **only** way around this that I can think of is to make an extension that replaces the Scratch project with the Phosphorus one and to inject the Phosphorus code into the project address. However, Nathan does not seem to be working on this anymore.

For the explanation of the second error:
```
Error: XHR Error
    at XMLHttpRequest.xhr.onerror
```
This is really just caused by the CORS error stated above. You can thank Scratch 3 for this.

**Comment by GarboMuffin** [comment]
Just want to say:

That first error is only phosphorus attempting to grab a project's title. That request can fail and nothing bad happens. The second error, however, happens because the location of new Scratch projects changed after Scratch 3. Simply updating the path wouldn't fix anything because, as you said, phosphorus cannot parse Scratch 3 projects. The new API route can actually be fetched by phosphorus without any weird extensions.

I really hate to advertise, but [I did actually Scratch 3 projects work in my phosphorus fork.](https://forkphorus.github.io/) It works... actually pretty well on some projects.

**Comment by IceHacks** [comment]
Yeah I misunderstood (I only briefly looked at the errors), because I don't really use it.
